libs/partners/milvus/langchain_milvus/vectorstores/milvus.py

```diff
@@ -128,6 +128,33 @@ class Milvus(VectorStore):
         metadata_field (str): Name of the metadta field. Defaults to None.
             When metadata_field is specified,
             the document's metadata will store as json.
+        field_schema  (Optional[dict]): What is the dataType of fields,
+            Default is Varchar, Example of field schema dict is :-
+            [
```

Comment:
@rgupta2508 I left some comments on your code, I think it is easy for you to optimize the code. And it seems we have to wait and rely on pymilvus to solve the [array-infer problem](https://github.com/milvus-io/pymilvus/pull/2144)

Reply:
Hi @zc277584121  ,
Thanks for review, i have incorporated the reviews comments.

Not sure if we should block langchain user to create array fields, metadata schema works in case of pymilvus returns other datatype. priority to create schema using metadata_schema field if metadata schema not available we will consider infer datatype from milvus.

One more point is,  If pymilvus return Array datatype in infer_dtype_bydata then also we need metadata schema to create collection, because for array Datatype we need some more mandatory parameter (ex:- max_capacity, max_length) , that can be added using metadata_schema.

Metadata schema can be used scalar index also,. currently milvus langchain only support indexing for _vector_field field.
https://github.com/langchain-ai/langchain/issues/24343